internal/cmd/marketplace/apply_test.go

```diff
@@ -0,0 +1,438 @@
+				"tenantId":            "team-rocket-test",
+				"type":                "plugin",
+			},
+		},
+	}
+
+	t.Run("should return response when is 200 OK", func(t *testing.T) {
+		mockResponse := &ApplyResponse{
+			Done: true,
+			Items: []ApplyResponseItem{
+				{
+					ItemID:   "some-id",
+					Name:     mockResName,
+					Done:     true,
+					Inserted: true,
+					Updated:  false,
+				},
+			},
+		}
+		server := applyMockServer(t, http.StatusOK, mockResponse)
+		defer server.Close()
+		clientConfig := &client.Config{
+			Transport: http.DefaultTransport,
+		}
+		clientConfig.Host = server.URL
+		client, err := client.APIClientForConfig(clientConfig)
+		require.NoError(t, err)
+
+		found, err := applyMarketplaceResource(
+			context.Background(),
+			client,
+			mockTenantID,
+			validReqMock,
+		)
+
+		require.NoError(t, err)
+		require.Equal(t, found, mockResponse)
+	})
+	t.Run("should return err if response is a http error", func(t *testing.T) {
+		mockResponse := map[string]interface{}{
+			"message":    "You are not allowed to perform the request!",
+			"statusCode": http.StatusForbidden,
+			"error":      "Forbidden",
+		}
+		server := applyMockServer(t, http.StatusForbidden, mockResponse)
+		defer server.Close()
+		clientConfig := &client.Config{
+			Transport: http.DefaultTransport,
+		}
+		clientConfig.Host = server.URL
+		client, err := client.APIClientForConfig(clientConfig)
+		require.NoError(t, err)
+
+		found, err := applyMarketplaceResource(
+			context.Background(),
+			client,
+			mockTenantID,
+			validReqMock,
+		)
+
+		require.EqualError(t, err, "You are not allowed to perform the request!")
+		require.Nil(t, found)
+	})
+}
+
+func TestApplyPrintApplyOutcome(t *testing.T) {
+	t.Run("should match snapshot with both valid files and validation errors", func(t *testing.T) {
+		mockOutcome := &ApplyResponse{
+			Done: false,
+			Items: []ApplyResponseItem{
+				{
+					ItemID:           "id1",
+					Name:             "some name 1",
+					Done:             true,
+					Inserted:         false,
+					Updated:          true,
+					ValidationErrors: []ApplyResponseItemValidationError{},
+				},
+				{
+					ItemID:           "id2",
+					Name:             "some name 2",
+					Done:             true,
+					Inserted:         true,
+					Updated:          false,
+					ValidationErrors: []ApplyResponseItemValidationError{},
+				},
+				{
+					ItemID:           "id3",
+					Name:             "some name 3",
+					Done:             true,
+					Inserted:         true,
+					Updated:          false,
+					ValidationErrors: []ApplyResponseItemValidationError{},
+				},
+				{
+					ItemID:   "id4",
+					Name:     "some name 4",
+					Done:     false,
+					Inserted: false,
+					Updated:  false,
+					ValidationErrors: []ApplyResponseItemValidationError{
+						{
+							Message: "some validation error",
+						},
+					},
+				},
+			},
+		}
+		snaps.MatchSnapshot(t, buildOutcomeSummaryAsTables(mockOutcome))
```

Comment:
I'm not sold on these tests.

Once, adding a test that check that a user output is perfectly equals can be uselessly fragile. If in future we change library that will use tabs for separating elements instead of spaces, we have to update these tests.
If we are testing that some key element are present then is better to check only for their presence via a regex.

The snap approach for testing a string equality seems a little excessive, we can reach it via some constant strings and `assert.Equal`.

Reply:
Considering that we're checking an entire e2e output of a command I think that the fragility of the test is acceptable.
Also because it serves its purpose if changing a library we obtain a different output.
In this case we're not checking only the presence of an information but also its output form.

If we `assert.Equal` the result we're doing almost the same thing without the benefit of having a "fast-way" to check the expected result

Reply:
Absolutely!